### ISSUE TITLE
Review isDataTypeFile,  isDataTypeBinary in all generators

### ISSUE THREAD

**Issue by wing328** [swagger-api/swagger-codegen#5103]
##### Description

Similar to https://github.com/swagger-api/swagger-codegen/issues/4764, we'll need to review other generators to make sure `isDataTypeFile` and `isDataTypeBinary` are properly overridden.

##### Swagger-codegen version

Latest master

##### Suggest a Fix

The fix should be similar to https://github.com/swagger-api/swagger-codegen/pull/5065

If anyone wants to help on this, please reply to let us know.

**Comment by ePaul** [comment]
Do we have a test case for this in our petstore yaml?

**Comment by wing328** [comment]
@ePaul yes, Petstore covers what we need for day 1 but now the project has grown a lot and basic tests cases in Petstore won't cover all the edge cases we've discovered so far.

Ideally, we should be able to leverage one of the server generators (e.g. Rails, Spring, etc) to generate server code based on a test spec covering all the edge cases, and use this to complement what Petstore lacks.
